addons/l10n_es_edi_tbai/models/account_edi_format.py

```diff
@@ -0,0 +1,498 @@
+        xsd_attachment = self.env.ref(xsd_id, False)
+        if xsd_attachment:
+            try:
+                L10nEsTbaiXmlUtils._validate_format_xsd(xml, xsd_id)
+            except UserError as e:
+                return {
+                    'error': str(e).split('\\n'),
+                    'blocking_level': 'error',
+                    'success': False,
+                }
+        return {}
+
+    # -------------------------------------------------------------------------
+    # TBAI XML BUILD
+    # -------------------------------------------------------------------------
+
+    def _l10n_es_tbai_get_invoice_xml(self, invoice, cancel=False):
+        # If peviously generated XML was posted and not rejected (success or timeout), reuse it
+        doc = invoice._get_l10n_es_tbai_submitted_xml(cancel)
+        if doc is not None:
+            return doc
+
+        # Otherwise, generate a new XML
+        values = {
+            'is_emission': not cancel,
+            'datetime_now': datetime.now(tz=timezone('Europe/Madrid')),
+            'datetime_strftime': datetime.strftime,
+            'timezone_eus': timezone,
+            'float_repr': partial(float_repr, precision_digits=2),
+        }
+        values.update(self._l10n_es_tbai_get_header_values(invoice))
+        values.update(self._l10n_es_tbai_get_subject_values(invoice, cancel))
+        values.update(self._l10n_es_tbai_get_invoice_values(invoice, cancel))
+        values.update(self._l10n_es_tbai_get_trail_values(invoice, cancel))
+        template_name = 'l10n_es_edi_tbai.template_invoice_main' + ('_cancel' if cancel else '_post')
+        xml_str = self.env.ref(template_name)._render(values)
+        xml_doc = L10nEsTbaiXmlUtils._cleanup_xml_content(xml_str)
+        xml_doc = self._l10n_es_tbai_sign_invoice(invoice, xml_doc)
+
+        # Store the XML as attachment to ensure it is never lost (even in case of timeout error)
+        invoice._update_l10n_es_tbai_submitted_xml(xml_doc=xml_doc, cancel=cancel)
+
+        return xml_doc
+
+    def _l10n_es_tbai_get_header_values(self, invoice):
+        return {
+            'tbai_version': L10N_ES_EDI_TBAI_VERSION
+        }
+
+    def _l10n_es_tbai_get_subject_values(self, invoice, cancel):
+        # === SENDER (EMISOR) ===
+        sender = invoice.company_id if invoice.is_sale_document() else invoice.commercial_partner_id
+        values = {
+            'sender_vat': sender.vat[2:] if sender.vat.startswith('ES') else sender.vat,
+            'sender': sender,
+        }
+        if cancel:
+            return values  # cancellation invoices do not specify recipients (they stay the same)
+
+        # TODO TicketBai supports simplified invoices WITH recipients: use checkbox in invoice ?
+        # Note that credit notes for simplified invoices are ALWAYS simplified BUT can have a recipient even if invoice doesn't
+        partner = invoice.commercial_partner_id if invoice.is_sale_document() else invoice.company_id
```

Comment:
you are managing only ('out_invoice', 'out_refund'), right?

Reply:
Yes, removed useless condition